src/tests/src/solver/optim-model-filler/test-component-filler-to-master-pb.cpp

```diff
@@ -0,0 +1,339 @@
+                                    unsigned varIndex,
+                                    const std::string objectiveId,
+                                    Config::Location locoation)
+{
+    auto varNode = nodeRegistry.create<Nodes::VariableNode>(varId,
+                                                            varIndex,
+                                                            TimeIndex::VARYING_IN_TIME_ONLY);
+    NodeRegistry node_registry(varNode, std::move(nodeRegistry));
+    Expression expression("expr-" + objectiveId, std::move(node_registry));
+    Objective objective(objectiveId, std::move(expression), locoation);
+    return objective;
+}
+
+class TwoObjsCreator_OneSubPb_OneMaster: public ObjectivesCreator
+{
+    using ObjectivesCreator::ObjectivesCreator;
+
+public:
+    std::vector<Objective> create() override
+    {
+        std::vector<Objective> objectives;
+        auto obj_1 = makeObjectiveFromVariable(nodeRegistry_,
+                                               "var-1",
+                                               0,
+                                               "obj-1",
+                                               Config::Location::SUBPROBLEMS);
+        auto obj_2 = makeObjectiveFromVariable(nodeRegistry_,
+                                               "var-2",
+                                               1,
+                                               "obj-2",
+                                               Config::Location::MASTER);
+
+        objectives.emplace_back(std::move(obj_1));
+        objectives.emplace_back(std::move(obj_2));
+        return objectives;
+    }
+};
+
+class TwoSubPbObjsCreator: public ObjectivesCreator
+{
+    using ObjectivesCreator::ObjectivesCreator;
+
+public:
+    std::vector<Objective> create() override
+    {
+        std::vector<Objective> objectives;
+        auto obj_1 = makeObjectiveFromVariable(nodeRegistry_,
+                                               "var-1",
+                                               0,
+                                               "obj-1",
+                                               Config::Location::SUBPROBLEMS);
+        auto obj_2 = makeObjectiveFromVariable(nodeRegistry_,
+                                               "var-2",
+                                               1,
+                                               "obj-2",
+                                               Config::Location::SUBPROBLEMS);
+
+        objectives.emplace_back(std::move(obj_1));
+        objectives.emplace_back(std::move(obj_2));
+        return objectives;
+    }
+};
+
+struct FactoryFixture
+{
+    FactoryFixture():
+        linear_pb(false, "sirius"),
+        optimEntityContainer(linear_pb, &dummy_data, &scenario_group_repo)
+    {
+    }
+
+    void initialize(std::unique_ptr<VariablesCreator> varCreator,
+                    std::unique_ptr<ObjectivesCreator> objCreator)
```

Comment:
Add template parameters, remove `initialize` method

```suggestion
template<class VariablesCreator, class ObjectivesCreator>
struct FactoryFixture
{
    FactoryFixture():
        linear_pb(false, "sirius"),
        optimEntityContainer(linear_pb, &dummy_data, &scenario_group_repo)
    {
    }

    void initialize(std::unique_ptr<VariablesCreator> varCreator,
                    std::unique_ptr<ObjectivesCreator> objCreator)
```